src/frontends/paddle/src/op/group_norm.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "default_opset.hpp"
+#include "openvino/frontend/paddle/node_context.hpp"
+#include "openvino/frontend/paddle/visibility.hpp"
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+template <typename T>
+std::vector<T> get_monotonic_range(T end_value, T start_value = T{0}, T step = T{1}) {
+    auto value_count = static_cast<std::size_t>(std::floor((end_value - start_value) / step));
+    std::vector<T> range(value_count);
+    size_t n = start_value - step;
+    std::generate(std::begin(range), std::end(range), [&n, &step]() -> T {
+        return n += step;
+    });
+    return range;
+}
+
+std::shared_ptr<ov::Node> get_monotonic_range_along_node_rank(const Output<ov::Node>& value,
+                                                              int64_t start_value,
+                                                              int64_t step) {
+    if (value.get_partial_shape().rank().is_static()) {
+        const auto range_value =
+            op::get_monotonic_range<int64_t>(value.get_partial_shape().rank().get_length(), start_value, step);
+        return default_opset::Constant::create(element::i64, {range_value.size()}, range_value);
+    }
+    const auto value_shape = std::make_shared<default_opset::ShapeOf>(value);
```

Comment:
Yes, Range path should be removed because [your algorithm](https://github.com/openvinotoolkit/openvino/blob/7bbfd3f182315beea5debb936ba185011cdfcedc/src/frontends/paddle/src/op/group_norm.cpp#L64) already requires the rank is static.

Reply:
updated